code/_globalvars/lists/cables.dm

```diff
@@ -0,0 +1,74 @@
+GLOBAL_LIST_INIT(cable_colors, list(
+	"blue" = COLOR_STRONG_BLUE,
+	"cyan" = COLOR_CYAN,
+	"green" = COLOR_DARK_LIME,
+	"orange" = COLOR_MOSTLY_PURE_ORANGE,
+	"pink" = COLOR_LIGHT_PINK,
+	"red" = COLOR_RED,
+	"white" = COLOR_WHITE,
+	"yellow" = COLOR_YELLOW
+	))
+
+#define CABLE_NORTH 1
+#define CABLE_SOUTH 2
+#define CABLE_EAST 4
+#define CABLE_WEST 8
+#define CABLE_NORTHEAST 16
+#define CABLE_SOUTHEAST 32
+#define CABLE_NORTHWEST 64
+#define CABLE_SOUTHWEST 128
+
+GLOBAL_LIST_INIT(cable_dirs, list(
+	CABLE_NORTH,
+	CABLE_SOUTH,
+	CABLE_EAST,
+	CABLE_WEST,
+	CABLE_NORTHEAST,
+	CABLE_SOUTHEAST,
+	CABLE_NORTHWEST,
+	CABLE_SOUTHWEST,
+	))
+
+GLOBAL_LIST_INIT(cable_dirs_to_inverse, list(
+	"[CABLE_NORTH]" = CABLE_SOUTH,
+	"[CABLE_SOUTH]" = CABLE_NORTH,
+	"[CABLE_EAST]" = CABLE_WEST,
+	"[CABLE_WEST]" = CABLE_EAST,
+	"[CABLE_NORTHEAST]" = CABLE_SOUTHWEST,
+	"[CABLE_SOUTHEAST]" = CABLE_NORTHWEST,
+	"[CABLE_NORTHWEST]" = CABLE_SOUTHEAST,
+	"[CABLE_SOUTHWEST]" = CABLE_NORTHEAST,
+	))
+
+GLOBAL_LIST_INIT(cable_dirs_to_real_dirs, list(
+	"[CABLE_NORTH]" = NORTH,
+	"[CABLE_SOUTH]" = SOUTH,
+	"[CABLE_EAST]" = EAST,
+	"[CABLE_WEST]" = WEST,
+	"[CABLE_NORTHEAST]" = NORTH|EAST,
+	"[CABLE_SOUTHEAST]" = SOUTH|EAST,
+	"[CABLE_NORTHWEST]" = NORTH|WEST,
+	"[CABLE_SOUTHWEST]" = SOUTH|WEST,
+	))
+
+GLOBAL_LIST_INIT(real_dirs_to_cable_dirs, list(
+	"[NORTH]" = CABLE_NORTH,
+	"[SOUTH]" = CABLE_SOUTH,
+	"[EAST]" = CABLE_EAST,
+	"[WEST]" = CABLE_WEST,
+	"[NORTH|EAST]" = CABLE_NORTHEAST,
+	"[SOUTH|EAST]" = CABLE_SOUTHEAST,
+	"[NORTH|WEST]" = CABLE_NORTHWEST,
+	"[SOUTH|WEST]" = CABLE_SOUTHWEST,
+	))
+
+GLOBAL_LIST_INIT(cable_dir_rotate_clockwise, list(
+	"[CABLE_NORTH]" = CABLE_EAST,
+	"[CABLE_SOUTH]" = CABLE_WEST,
+	"[CABLE_EAST]" = CABLE_SOUTH,
+	"[CABLE_WEST]" = CABLE_NORTH,
+	"[CABLE_NORTHEAST]" = CABLE_SOUTHEAST,
+	"[CABLE_SOUTHEAST]" = CABLE_SOUTHWEST,
+	"[CABLE_NORTHWEST]" = CABLE_NORTHEAST,
+	"[CABLE_SOUTHWEST]" = CABLE_NORTHWEST,
+	))
```

Comment:
this is going to sound stupid but lists are pretty small in memory footprint, so you can pre-alloc the lists to 128 using `new /list(128)`. Associative list access is O(log N), numeric access is O(1) because it's just a pointer.

Additionally, you can use `GLOBAL_REAL_VAR` instead of bothering with GLOB, it saves on the datum var access which can matter at this scale.

Not a required change just food for thought.